x/icacallbacks/module_ibc.go

```diff
@@ -93,30 +94,70 @@ func (im IBCModule) NegotiateAppVersion(
 	return proposedVersion, nil
 }
 
-// GetTxMsgData returns the msgs from an ICA transaction and can be reused across authentication modules
-func GetTxMsgData(ctx sdk.Context, ack channeltypes.Acknowledgement, logger log.Logger) (*types.ICATxResponse, error) {
+// UnpackAcknowledgementResponse returns the msgs from an ICA transaction and can be reused across authentication modules
+func UnpackAcknowledgementResponse(ctx sdk.Context, ack []byte, logger log.Logger) (*types.ICATxResponse, error) {
+	// NOTE: Is this safe to assume? According to the ics-27 spec, errors on host chains should be reflected in the acknowledgement type
+	// so Acknowledgement_Result should only be returned if the transaction was successful
+	// However, I recall seeing a case where the acknowledgement type was Acknowledgement_Result but the transaction failed
+	//
+	// -> If my memory's correct, I think a failure on the host could come back as either an AcknowledgementError
```

Comment:
also recall this - I think the way we tested it (as you mentioned below) was delegating to a non-existent validator?